core/frontend/src/render/webgl/glsl/Clipping.ts

```diff
@@ -66,6 +67,54 @@ const applyClipPlanesPostlude = `
     } else if (!clippedByCurrentPlaneSet && calcClipPlaneDist(v_eyeSpace, plane) < 0.0) {
       clippedByCurrentPlaneSet = true;
     }
+
+    //avoiding near and far clip planes
+    if (i <= u_clipParams[1] - 2) {
+
+      /* 
+      * The closest point on a plane from a point, p,  will be: p minus the distance between p and the plane, in the direction of the plane's normal vector.
+      * We have normal as plane.xyz, and our point as v_eyeSpace.
+      * We can find the distance from the plane using calcClipPlaneDist, then multiply that by the plane's normal.
+      * Subtract the result from the original point to obtain the location of the closest point on the clip plane to v_eyeSpace.
+      * Finally, convert the point on the plane to window coords, and take the distance between that and gl_FragCoord,
+      * to determine whether or not to highlight the current fragment. 
+      */
+
+      float d = calcClipPlaneDist(v_eyeSpace, plane);
+      vec4 pointOnPlane = vec4((v_eyeSpace - (d * plane.xyz)), 1.0);
+
+      //Converting original point to window coords
+      //Only doing this now to ensure the conversion is correct by later comparing with gl_FragCoord
+      //In final implementation, pointOnPlane will be converted, not v_eyeSpace
+
+      vec4 convertedEyeSpace = vec4(v_eyeSpace, 1.0);
+      convertedEyeSpace = convertedEyeSpace * u_proj;
+      convertedEyeSpace.xyz /= convertedEyeSpace.w;                               //Should now be in NDC
+
+      //This will convert from NDC to window coords
+      //commented out for now to check previous conversion
+      //convertedEyeSpace.x = ((convertedEyeSpace.x + 1.0) * 0.5 * u_viewport.x);
+      //convertedEyeSpace.y = ((convertedEyeSpace.y + 1.0) * 0.5 * u_viewport.y);   //Should now be in window coords
+
+      //This check will show if it is correctly in NDC
+      //since NDC is in range -1 to 1, we should expect every point to be red
+      // while being darkest at the center, getting brighter as absolute value of x increases
+      if ((abs(convertedEyeSpace.x) <= 1.0) && (abs(convertedEyeSpace.y) <= 1.0) && (!clippedByCurrentPlaneSet)) {
+        g_clipColor = vec3(abs(convertedEyeSpace.x), 0.0, 0.0);
+        return true;
+      }
+
+
+      // A possible workaround if we can't get u_proj to work correctly is to do something like this
+      // By dividing by v_eyeSpace.z, we can get a d that changes as we zoom in and out
+      // Which will emulate the behavior of a purely pixel based distance check
+
+      // float d = abs(calcClipPlaneDist(v_eyeSpace, plane) / v_eyeSpace.z);
```

Comment:
https://github.com/iTwin/itwinjs-core/assets/101671244/182498fb-1d13-40e8-a169-945ab273edc1

Here is a clip of what this looks like in action, it generally does emulate the behavior of a true conversion.